Tutorials: XGI in X minutes

thomasrobiglio opened the pull request:
I plan on working on this PR related to #414 over the next two weeks.
(quick note: the texts rely heavily on chatGPT's joy of life, we can then decide to tone it down a bit 😄)

maximelucas commented on tutorials/XGI in 5 minutes.ipynb:
```diff
@@ -0,0 +1,477 @@
+{
```
<p>There are more than 2 options: you can also read from file or load a dataset. I would just say "in XGI in 1 min we saw option 1 (describe), now we're using a second option (describe)"</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/xgi-org/xgi/pull/415/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='4'/>

maximelucas replied:
<p>I would put this before the circular layout, so that the user can immediately check that the plotted structure corresponds to the output of <code>H.edges.members()</code>. Also I'm wondering if we should use fixed seed to keep the same layout the three times that we use <code>xgi.draw()</code></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/xgi-org/xgi/pull/415/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='15'/>

maximelucas replied:
<p>nice! i think we should have only integer xticks, and also enforce that the bins are centered on the integers. also maybe add "# occurrences" for the ylabel</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/xgi-org/xgi/pull/415/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='23'/>

maximelucas replied:
<p>nice. i would just add a ylabel</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/xgi-org/xgi/pull/415/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='25'/>

maximelucas replied:
<p>maybe add a link to where people can find out about other generators</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/xgi-org/xgi/pull/415/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='26'/>

maximelucas commented on tutorials/XGI in 5 minutes.ipynb:
```diff
@@ -0,0 +1,477 @@
+{
```
<p>we should try to update the circular layout function so that it's actually a circle, not distorted</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/xgi-org/xgi/pull/415/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='11'/>

thomasrobiglio replied:
We can open an issue about that, but do we want that or not? in networkx they don't have it (see example)

<img width="810" alt="Screenshot 2023-07-23 alle 18 42 01" src="https://github.com/xgi-org/xgi/assets/83019028/f2e764ef-ffe2-4fdc-ad03-0d7e17f12d75">

maximelucas replied:
Let's see, I opened an issue.

thomasrobiglio commented:
Thank you @maximelucas I will address your comments in the next days while working on the 15 min tuto

maximelucas commented on tutorials/XGI in 15 minutes.ipynb:
```diff
@@ -0,0 +1,333 @@
+{
```
<p>-&gt; "with which you can easily"</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/xgi-org/xgi/pull/415/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

maximelucas replied:
<p>we already mentioned the datasheet above</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/xgi-org/xgi/pull/415/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='8'/>

maximelucas replied:
<p>-&gt; connected</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/xgi-org/xgi/pull/415/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='13'/>

maximelucas replied:
<p>comment on how many nodes and edges were removed?</p><p>also maybe before cleaning, get the number of isolates/singletons and see that the final count of the clean H is consistent?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/xgi-org/xgi/pull/415/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='16'/>

maximelucas replied:
<p>the XGI -&gt; XGI</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/xgi-org/xgi/pull/415/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='19'/>

maximelucas replied:
<p>I would replace </p><p> </p><pre class="ql-syntax" spellcheck="false">plt.figure(figsize=(10, 10))
ax = plt.subplot(111)
</pre><p> </p><p>by one line</p><p> </p><pre class="ql-syntax" spellcheck="false">fig, ax = plt.subplots((figsize=(10, 10))
</pre>
<br/>

 _Reply via <a href='https://app.reviewnb.com/xgi-org/xgi/pull/415/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='20'/>

maximelucas replied:
<p>also maybe before plotting stats, plot the default viz?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/xgi-org/xgi/pull/415/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='21'/>

maximelucas commented:
As Leo mentioned somewhere, we should check that the Github Action testing the notebook will test the new ones (it might test only names "Tutorial_*").

Quickly look at [it](https://github.com/xgi-org/xgi/blob/main/.github/workflows/test.yml) though, it though looks fine?

thomasrobiglio commented:
Ok, the three tutorials are ready, I have already addressed the comments by @maximelucas. As I had mentioned on Zulip maybe we can merge this before releasing the new version and then proceed with the rest of the tutorial restructuring (that's why I have renamed the PR and unlinked #414) later.

maximelucas commented on tutorials/XGI in 15 minutes.ipynb:
```diff
@@ -0,0 +1,679 @@
+{
```
<p>Ref to XGI in 5 min still missing :)</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/xgi-org/xgi/pull/415/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='0'/>

maximelucas replied:
<p>Remove ' for example" ', at the beginning of the sentence</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/xgi-org/xgi/pull/415/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='4'/>

maximelucas replied:
<p>After this I would just say something like "For other hypergraph matrices such as Laplacians, see the linalg documentation" with a link.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/xgi-org/xgi/pull/415/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='37'/>

maximelucas replied:
<p>Before wrapping up (or earlier in the tuto), what do you think of just advertising the stats with 2 short examples, and the link to docs/tuto?</p><p> </p><p>Like </p><ul><li>H.nodes.filterby("degree", 2)</li><li>df = H.nodes.multi(["degree", "clustering_coefficient"]).aspandas()</li></ul>
<br/>

 _Reply via <a href='https://app.reviewnb.com/xgi-org/xgi/pull/415/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='42'/>

thomasrobiglio commented:
thank you @maximelucas for the comments, everything should be good to go now from my side

maximelucas commented:
Looks good, thanks Thomas!!
